gdshelpers/parts/waveguide.py

```diff
@@ -85,25 +85,16 @@ def get_segments(self):
         """
         return [(port.copy(), obj, length) for port, obj, length in self._segments]
 
-    def add_straight_segment(self, length, final_width=None, **kargs):
-        self._current_port.set_port_properties(**kargs)
-        final_width = final_width or self.width
-
-        endpoint = shapely.geometry.Point(length, 0)
+    def add_straight_segment(self, length, final_width=None, **kwargs):
+        self._current_port.set_port_properties(**kwargs)
+        final_width = final_width if final_width is not None else self.width
 
         if not np.isclose(length, 0):
             assert length >= 0, 'Length of straight segment must not be negative'
-            polygon = shapely.geometry.Polygon(([0, self.width / 2], [length, final_width / 2],
-                                                [length, -final_width / 2], [0, -self.width / 2]))
-            polygon = shapely.affinity.rotate(polygon, self.angle, origin=[0, 0], use_radians=True)
-            polygon = shapely.affinity.translate(polygon, self.x, self.y)
-            self._segments.append((self._current_port.copy(), polygon, length))
-
-        endpoint = shapely.affinity.rotate(endpoint, self.angle, origin=[0, 0], use_radians=True)
-        endpoint = shapely.affinity.translate(endpoint, self.x, self.y)
 
-        self._current_port.origin = endpoint.coords[0]
-        self._current_port.width = final_width
+            self.add_parameterized_path(path=lambda t: [t * length, 0],
+                                        width=lambda t: np.array(self.width) * (1 - t) + np.array(final_width) * t,
```

Comment:
In the current design, self.width is always an array, so `np.array(self.width)` is not necessary

Reply:
See above, now self.width is only an array if it has more than one element. Therefore, we could remove `np.array(...)` here, as anyway only tapering between elements with the same number of rails/gaps is allowed (?)